packages/workflow-compile/legacy/index.js

```diff
@@ -92,18 +92,53 @@ const Contracts = {
         const compile = SUPPORTED_COMPILERS[compiler];
         if (!compile) throw new Error("Unsupported compiler: " + compiler);
 
-        const compileFunc = multiPromisify(
-          config.all === true || config.compileAll === true
-            ? compile.all
-            : compile.necessary
-        );
-
-        let [contracts, output, compilerUsed] = await compileFunc(config);
+        config.compilersInfo = [];
+        let contracts, output, compilerUsed;
+        if (compiler === "solc") {
+          const compileFunc = multiPromisify(
+            config.all === true || config.compileAll === true
+              ? compile.all
+              : compile.necessary
+          );
+
+          const result = await compileFunc(config);
+
+          contracts = result[0];
+          output = result[1];
+          compilerUsed = result[2];
+        } else {
+          const { Compile } = compile;
+          const compileFunc =
+            config.all === true || config.compileAll === true
+              ? Compile.all
+              : Compile.necessary;
+
+          const { compilations } = await compileFunc(config);
+          const result = compilations.reduce(
+            (a, compilation) => {
+              for (const contract of compilation.contracts) {
+                a.contracts[contract.contractName] = contract;
+              }
+              a.output = a.output.concat(compilation.sourceIndexes);
+              return a;
+            },
+            {
+              contracts: {},
+              output: []
+            }
+          );
+          contracts = result.contracts;
+          output = result.output;
+          if (compilations[0] && compilations[0].compiler) {
+            compilerUsed = {
+              name: compilations[0].compiler.name,
+              version: compilations[0].compiler.version
+            };
+          }
```

Comment:
Should we use the `NewToLegacy` shims for these?

Reply:
I don't think those shims get us what we want, do they? The only shims in there are the shims for a contract and the bytecode.

Reply:
Don't we need to shim the bytecode?

Reply:
Oh, I suppose you are right about that. Doh!